geoh5py/shared/concatenation/concatenator.py

```diff
@@ -616,3 +617,186 @@ def update_array_attribute(
             self.data[alias] = values
 
         self.save_attribute(field)
+
+    def _extract_objects_data_table(
+        self, data_name: str
+    ) -> tuple[np.ndarray, ConcatenatedPropertyGroup, Any]:
+        """
+        Extract the data table for a given data name.
+        This function is an internal function used by the get_data_table function.
+
+        :param data_name: The name of the data to extract.
+
+        :return: The index, the property group and the NDV value of the data
+        """
+        # ensure data_name is in the data
+        if data_name not in self.data:
+            raise KeyError(f"Data '{data_name}' not found in concatenated data.")
+
+        # get the index of the data and sort it
+        data_index = np.sort(self.index[data_name], order="Start index")
+
+        # get the first object of the know association
+        object_: ConcatenatedObject = self.workspace.get_entity(  # type: ignore
+            UUID(data_index[0][-2].decode("utf-8").strip("{}"))
+        )[0]
+
+        data: Data = object_.get_data(
+            UUID(data_index[0][-1].decode("utf-8").strip("{}"))
+        )[0]
+
+        # ensure the association is Depth:
+        if data.association != DataAssociationEnum.DEPTH:
+            raise TypeError(
+                f"Data '{data_name}' is not associated with depth ({data.association})."
+            )
+
+        # define NDV value
+        ndv = np.nan
+        if isinstance(data, NumericData):
+            ndv = data.nan_value
+        elif isinstance(data, TextData):
+            ndv = ""
+
+        property_group: ConcatenatedPropertyGroup = getattr(data, "property_group")
+
+        return data_index, property_group, ndv
+
+    def _get_association_data_table(
+        self, property_group: ConcatenatedPropertyGroup
+    ) -> np.ndarray:
+        """
+        Get the association table for a given property group.
+        This function is an internal function used by the get_data_table function.
+
+        :param property_group: The property group to extract.
+
+        :return: the association table of the property group.
+        """
+        # get the associations
+        associations: np.ndarray = np.array([])
+        # todo: if association is DEPTH,. it just can be "Interval Table" or "Depth Table"
+        if property_group.property_group_type == "Interval table":
+            associations = np.array(
+                [
+                    [
+                        property_group.from_.name,
+                        np.sort(
+                            self.index[property_group.from_.name], order="Start index"
+                        ),
+                        self.data[property_group.from_.name],
+                    ],
+                    [
+                        property_group.to_.name,
+                        np.sort(
```

Comment:
Looks like you are assuming that from and to will be sorted the same? They might not always line up with the same drillholes

Reply:
Logic cahnged